### DISCUSSION TITLE
Get dynamic number of lines and width working with 1.7.0

### DISCUSSION THREAD

**Discussion by abhi3o** [davatorium/rofi#1455] [Theming] [answered]
I am using [this](https://github.com/zbaylin/rofi-wifi-menu/blob/master/rofi-wifi-menu.sh) rofi based wifi-menu script with a custom theme.

The script has this line, which just gives the length of first row adjusted for the biggest Wi-Fi name,
`RWIDTH=$(($(echo "$LIST" | head -n 1 | awk '{print length($0); }')+2))`

and this variable was passed to as like,
`rofi -dmenu -p "Wi-Fi SSID: " -lines "$LINENUM" -width -"$RWIDTH"`

but it doesn't work with rofi-1.7.0, resulting in a half-screen width menu. And the lines argument doesn't work too. Both of them worked in 1.6.1.

I have read about the new changes: `rofi -theme-str 'window {width: 50%;}'` and so changed that line to `rofi -dmenu -p "Wi-Fi SSID: " -theme-str 'window {width: "$RWIDTH"; lines: "$LINENUM";}'` but this doesn't change anything. I need character based width, so need to add '-' in front of the "$RWIDTH", but raises a parser error. Even the "$LINENUM" doesn't affect the shown menu.

How do I get it back working for dynamic number of lines and width based on characters/font again?

**Comment by DaveDavenport** [comment, marked as answer]
I think in the second commandline your shell escaping is wrong and the variables are not expanded by the shell and they are also not in the right format for rofi's environment support (see theme manpage)

**Reply by DaveDavenport** [reply]
dmenu mode in rofi still supports the `-l` dmenu compatibility option. (if you have a non-fullscreen theme) just like it did in previous versions.

Width is something that is part of the theme, changing the width of rofi from a script can seriously break the users theme and should therefor not be done. Unless it is part of the theme itself (Rofi 1.7.1 will have extra support for this).

**Reply by DaveDavenport** [reply]
See https://github.com/davatorium/rofi/discussions/1411

**Reply by abhi3o** [reply]
Yes, I tried with bash variable expansion, but can't get it to work (poor scripting skills :p).

rofi's environment variable support does work though, but only this `rofi -theme-str 'window {width: ${WIDTH};}'` works and not `rofi -theme-str 'window {width: env(WIDTH, 30ch);}'`

Thanks for the quick support.

**Reply by DaveDavenport** [reply]
Yes,  please read what I mentioned above,  extra support for this will be in 1.7.1.  This has not been released.

**Reply by abhi3o** [reply]
Yes, I saw that it needs testing. I read the rofi-theme man page from Github, instead of 1.7.0's!

Dynamic width is necessary for this script because if, say some Wi-Fi name is larger than the fixed width we set, then the other columns, like that of SIGNAL, SECURITY, etc. get chopped off.

**Reply by DaveDavenport** [reply]
maybe do some pre-processing where you ellipsize the wi-fi name.  Otherwise a very long wifi name can make your rofi width  larger then your monitor.
